pkg/scm/git/git.go

```diff
@@ -379,6 +395,33 @@ func doesExist(dir string) bool {
 	return !(err != nil && os.IsNotExist(err))
 }
 
+// isEmptyGitRepository checks if there are any commits or remotes tracked in
+// the repository, returns true if it is empty
+func isEmptyGitRepository(dir string) bool {
+	var isEmpty = true // The return value, true for empty
+
+	gitDir := strings.TrimPrefix(dir, "file://") + "/.git"
+	// Search the content of the .git directory for content
+	directories := [2]string{gitDir + "/objects", gitDir + "/refs"}
+
+	for _, dir := range directories {
+		filepath.Walk(dir,
+			func(path string, info os.FileInfo, err error) error {
+				if info != nil {
+					// If we find a file, the git directory is "not empty"
+					// We're looking for object blobs, and ref files
+					if !info.IsDir() {
+						isEmpty = false
```

Comment:
If we found a file do we need to continue or one file is enough? In this case we can put `break` here, right?